src/main/java/frc/robot/commands/auto/Balance.java

```diff
@@ -0,0 +1,99 @@
+package frc.robot.commands.auto;
+
+import com.ctre.phoenix.sensors.Pigeon2;
+import com.ctre.phoenix.sensors.PigeonIMU;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.RobotState;
+import frc.robot.subsystems.drive.DriveSubsystem;
+import frc.robot.subsystems.drive.PigeonInterface;
+
+public class Balance extends CommandBase {
+
+    private DriveSubsystem drive;
+    private PigeonInterface pigeon = new PigeonInterface();
+
+    private PIDController driveController, balanceController;
+
+    private double targetStationX = 0; //TODO: get location of targetStationX
+
+    private Timer searchTimer = new Timer();
+    private Timer balanceTimer = new Timer();
+
+    private boolean balanceFound;
+
+    //TODO: get pid values
+    private double kPDrive = 0, kIDrive = 0, kDDrive = 0, kPBalance = 0, kIBalance = 0, kDBalance = 0;
+
+    public Balance(DriveSubsystem drive) {
+        this.drive = drive;
+        driveController = new PIDController(kPDrive, kIDrive, kDDrive);
+        balanceController = new PIDController(kPBalance, kIBalance, kDBalance);
```

Comment:
This should be handled in DriveSubsystem.

You could create another mode in DriveSubsystem (perhaps named Balance) that carries out the balancing routine.